libtorch-ffi/libtorch-ffi.cabal

```diff
@@ -63,11 +63,18 @@ library
                     , safe-exceptions
                     , sysinfo
                     , async
- extra-libraries:     stdc++
-                    , c10
-                    , iomp5
-                    , mklml
-                    , torch
+ if os(darwin)
+  extra-libraries:     c++
+                     , c10
+                     , iomp5
+                     -- , mklml
+                     , torch
+ else
+  extra-libraries:     stdc++
+                     , c10
+                     , iomp5
+                     -- , mklml
```

Comment:
CI  but I don't know if we can delete it,
I'll add a test later to think about what to do.

Reply:
I can not find any dependency of libmklml.os.
So I think it is acceptable.
https://gist.github.com/junjihashimoto/f66a1e34ae8770a35573555e84c0bb2d

Reply:
macos' library-dependency is different from linux's one.
```
bash-3.2$ otool -L libtorch.dylib
libtorch.dylib:
        @rpath/libtorch.dylib (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate (compatibility version 1.0.0, current version 4.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
        @rpath/libmklml.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libiomp5.dylib (compatibility version 5.0.0, current version 5.0.0)
        @rpath/libc10.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
bash-3.2$ otool -L libc10.dylib
libc10.dylib:
        @rpath/libc10.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

Reply:
macos's library needs both libiomp5 and libmklml which linux does not need.

Reply:
great findings, thanks for looking into this @junjihashimoto! when I rewrote this cabal file, I was very conservative and kept any dependency I could fulfil with nix. It’s great that we can get rid of some!